### PR TITLE
DOC: update the docstring of pandas.DataFrame.from_dict

### PR DESCRIPTION
**Pull request by ReubenMarkham**
```
################################################################################
#################### Docstring (pandas.DataFrame.from_dict) ####################
################################################################################

Construct DataFrame from dict of array-like or dicts.

Creates DataFrame object from dictionary by columns or by index
allowing dtype specification.

Parameters
----------
data : dict
    Of the form {field : array-like} or {field : dict}.
orient : {'columns', 'index'}, default 'columns'
    The "orientation" of the data. If the keys of the passed dict
    should be the columns of the resulting DataFrame, pass 'columns'
    (default). Otherwise if the keys should be rows, pass 'index'.
dtype : dtype, default None
    Data type to force, otherwise infer.
columns : list, default None
    Column labels to use when orient='index'. Raises a ValueError
    if used with orient='columns'
    .. versionadded:: 0.23.0.

See Also
--------
DataFrame.from_records : Create DataFrame from ndarray (structured dtype),
                         list of tuples, dict, or DataFrame
DataFrame.from_items : Create DataFrame from sequence of (key, value) pairs.

Returns
-------
DataFrame

Examples
--------
>>> data = {'col_1': [3, 2, 1, 0], 'col_2': ['a', 'b', 'c', 'd']}
>>> data
{'col_1': [3, 2, 1, 0], 'col_2': ['a', 'b', 'c', 'd']}

>>> pd.DataFrame.from_dict(data)
   col_1 col_2
0      3     a
1      2     b
2      1     c
3      0     d

>>> pd.DataFrame.from_dict(data, orient='index')
       0  1  2  3
col_1  3  2  1  0
col_2  a  b  c  d

################################################################################
################################## Validation ##################################
################################################################################

Docstring for "pandas.DataFrame.from_dict" correct. :)
```

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -883,27 +883,53 @@ def dot(self, other):
...
             should be the columns of the resulting DataFrame, pass 'columns'
             (default). Otherwise if the keys should be rows, pass 'index'.
         dtype : dtype, default None
-            Data type to force, otherwise infer
-        columns: list, default None
+            Data type to force, otherwise infer.
+        columns : list, default None
             Column labels to use when orient='index'. Raises a ValueError
             if used with orient='columns'
+            .. versionadded:: 0.23.0.
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py]: You need to keep a blank line above this one (restructured text specific gotcha ..)

File: pandas/core/frame.py [hunk 2]
```diff
@@ -883,27 +883,53 @@ def dot(self, other):
...
 
-            .. versionadded:: 0.23.0
+        See Also
+        --------
+        DataFrame.from_records : DataFrame from ndarray (structured dtype),
+                                 list of tuples, dict, or DataFrame
+        DataFrame.from_items : DataFrame from sequence of (key, value) pairs.
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py]: Can you also add a link to simply 'DataFrame', as that can also be used as constructor

**Review by jorisvandenbossche — COMMENTED** [review]
Small comments, looking good!

File: pandas/core/frame.py (hunk 2 above)
Comment (jreback) [reply pandas/core/frame.py]: remove from_items as that is now deprecated.

File: pandas/core/generic.py [hunk 3]
```diff
@@ -5548,53 +5548,77 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
...
-
-        >>> df.clip(-1.0, 0.5)
-                  0         1
-        0  0.335232 -1.000000
-        1 -1.000000  0.500000
-        2  0.027753 -1.000000
-        3  0.230930 -0.679613
-        4  0.500000  0.500000
-
+           col_0  col_1
```
Comment (jreback) [inline pandas/core/generic.py]: this looks from another PR, can you revert.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by TomAugspurger** [comment]
@ReubenMarkham do want to continue working on this?

**Pull request closed without merging**